File: 645-set-mismatch/set-mismatch.cpp

class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        int xor1 = 0, xor2 = 0;

        for (int i = 0; i < n; ++i) {
            xor1 ^= nums[i];
        }

        for (int i = 1; i <= n; ++i) {
            xor2 ^= i;
        }
        int dup = -1;
        std::unordered_map<int, int> count;
        for (int num : nums) {
            count[num]++;
            if (count[num] == 2) {
                dup = num;
                break;
            }
        }

        int missing = xor1 ^ xor2 ^ dup;

        return {dup, missing};
    }
};

File: 645-set-mismatch/set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n=len(nums)
        xor1=0
        xor2=0
        for i in range(n):
            xor1^=nums[i]
        for i in range(1,n+1):
            xor2^=i
        for num in nums:
            if nums.count(num)==2:
                dup=num
                break
        missing=xor1^xor2^dup
        return [dup,missing]


        
            

        